nifi-marklogic-processors/src/main/java/org/apache/nifi/marklogic/processor/AbstractMarkLogicProcessor.java

```diff
@@ -262,8 +272,11 @@ protected void logError(Throwable t) {
     }
 
     /**
-     * See <a href="https://github.com/marklogic/nifi/issues/21">...</a> for details on why the calls to transfer and commit are within
-     * a synchronized block.
+     * The purpose of the synchronized block is to ensure that when a processor instance runs multiple threads - such
```

Comment:
Added this comment here as the referenced NiFi ticket didn't provide the detailed explanation that's now here. An internal Wiki page captures further details on this topic.